File: product-order-service/src/main/java/com/example/productorderservice/order/CreateOrderRequest.java
package com.example.productorderservice.order;

import org.springframework.util.Assert;

record CreateOrderRequest(Long productId, int quantity) {
    CreateOrderRequest {
        Assert.notNull(productId, "상품 ID는 필수입니다.");
        Assert.isTrue(quantity > 0, "수량은 0보다 커야 합니다.");
    }
}

File: distributed-transaction/saga-choreography-order/src/main/kotlin/org/example/order/controller/dto/CreateOrderRequest.kt
package org.example.order.controller.dto

import org.example.order.application.dto.CreateOrderCommand

data class CreateOrderRequest(
    val items: List<OrderItem>,
) {

    fun toCommand() = CreateOrderCommand(
        items.map { CreateOrderCommand.OrderItem(it.productId, it.quantity) }
    )


    data class OrderItem(
        val productId: Long,
        val quantity: Long,
    )
}
